core/src/main/java/com/netflix/conductor/core/execution/DeciderService.java

```diff
@@ -446,7 +449,10 @@ boolean isResponseTimedOut(TaskDef taskDefinition, Task task) {
         String taskId = IDGenerator.generate();
         TaskMapperContext taskMapperContext = new TaskMapperContext(workflowDefinition, workflowInstance, taskToSchedule,
                 input, retryCount, retriedTaskId, taskId, this);
-        return taskMappers.get(taskType.name()).getMappedTasks(taskMapperContext);
+        List<Task> tasks = taskMappers.get(taskType.name()).getMappedTasks(taskMapperContext);
+
+        return tasks.stream().filter(task -> executionDAO.getPendingTasksByWorkflow(task.getTaskType(), workflowInstance.getWorkflowId()).stream().
```

Comment:
please add a comment explaining why we need to filter tasks here

Reply:
added comment